src/components/Button/Button.tsx

```diff
@@ -52,6 +52,7 @@ interface CommonButtonProps
     ButtonProps,
     | 'id'
     | 'accessibilityLabel'
+    | 'role'
```

Comment:
Late to the party, but thoughts on triggering a warn if the `role` isn't in a defined list of `Button` appropriate roles in order to take away some foot guns?

Reply:
We could maintain an `enum` of roles, and then subset `enums` for particular HTML elements. This would mean TypeScript indicates when an invalid value is passed. But I'm not sure thats really the job of Polaris to validate something like this. The spec defining what values are available to `role` could be updated to include more options in the future, so the burden of maintaining that would be on Polaris.

Would be curious to hear what someone on Polaris thinks of this idea.